win32/tests/gballoc_hl_int/CMakeLists.txt

```diff
@@ -18,15 +18,18 @@ set(${theseTestsName}_test_files
 
 set(${theseTestsName}_c_files
     ../../src/gballoc_ll_${gballoc_ll_impl_lower}.c
-    ../../src/gballoc_hl_passthrough.c
+    ../../src/gballoc_hl_${gballoc_hl_impl_lower}.c
+    ../../src/timer.c #needed because gballoc_hl needs timer to compute "how much time it takes"
+    ../../../common/src/gballoc.c #because timer will allocate memory with gballoc... 
+    ../../src/lock_win32.c #because timer uses locking... 

```

Comment:
I think that all the int tests should link to the pal lib. Because (eventually) this int test should go in the interfaces folder and run on both Windows and Linux (of course just the subset of passthrough/crt might be the only combination that is supported on Linux).

Reply:
Agree! However... easier to say than done. Here's why. Linking against the lib means there 1 (one) library. Well - one library means "one implementation of ll/hl". Which means we'd have to build the azure-c-pal for EVERY combination of hl/ll and link the tests against that. Not impossible, but at this moment I would like to do that with a different task or else this PR never gets checked in.

Notice how the library is a different target than here, and to use it here, it would need to exist predefined before running the test. As is now, it keep the madness of "foreach/foreach" (and frankly there should be "foreach" for every mltiple choices option we have).


Here's a task that tracks linking int tests against the library and not against individual files: Task 7781632: link int tests against the library and not against individual files:


---
In reply to: [459665962](https://github.com/Azure/azure-c-pal/pull/14#discussion_r459665962) [](ancestors = 459665962)